### DISCUSSION TITLE
Add Table To Custom Page Filament

### DISCUSSION THREAD

**Discussion by MahmoudAdelMansour** [filamentphp/filament#12716] [Help]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

correctlyHi , i have created a page to list the carts that comes from orders , carts had a pivot table called cart_order
whatever i want to create a new page and list the cart_order or any data i want , 
I have read the documentation , and i just create a page 
```
art make:filament-page Carts
```
and now i have an empty page , i searched at the docs but all i found is about title and color and label , 
and if i understand correctly , the page created as live-wire component page to write a code into , 
so i can't do 
```
class Carts extends Page
{
    protected static ?string $navigationIcon = 'heroicon-o-document-text';

    protected static string $view = 'filament.admin.pages.carts';
    public static function table(Table $table): Table
    {
        return $table
            ->columns([


            ])
            ->filters([
                //
            ]);

    }
    public static function getNavigationLabel(): string
    {
        return __('carts_of_clients');
    }
}
```
so there's anyway to make a page with listing a table without create a resource ?
I'm also considering creating a resources page for a shopping cart and implementing an action to display the order details.

**Comment by leandrocfe** [comment]
Follow this section
https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component#adding-the-table

**Reply by fariqueparammel** [reply]
hey am a newbie to laravel and filament i tried the doc but
gets the error Table [App\Filament\App\Resources\ProjectsResource\Pages\ManualDataEntry] must have a [query()].

blade view
<x-filament-panels::page>
    {{-- use interactwithrecords to interact with db i think --}}
    {{-- {{ $getRecord()->name }} --}}
    {{ $this->table }}
    {{ $this->form }}
</x-filament-panels::page>




resource\projectResource\page\ManualDataEntry.php
<?php

namespace App\Filament\App\Resources\ProjectsResource\Pages;

use App\Filament\App\Resources\ProjectsResource;
use Filament\Forms\Components\KeyValue;

use Filament\Resources\Pages\Page;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Tables\Columns\TextColumn;
use Filament\Tables\Concerns\InteractsWithTable;
use Filament\Tables\Contracts\HasTable;
use Filament\Tables\Table;
use App\Models\Projects;

class ManualDataEntry extends Page implements HasForms, HasTable
{
    use InteractsWithTable;
    use InteractsWithForms;

    protected static string $resource = ProjectsResource::class;

    protected static string $view = 'filament.app.resources.projects-resource.pages.manual-data-entry';
    protected ?string $heading = " ";
    protected static ?string $title = 'Enter Data';
  public function table(Table $table): Table
    {
        return $table
            ->emptyStateHeading('No posts yet')
            ->columns([

                TextColumn::make('currency')
                    ->money('EUR')
            ])
            ->filters([
                // ...
            ])
            ->actions([
                // ...
            ])
            ->bulkActions([
                // ...
            ]);
    }
    

// i want to dynamically add contents including column heading to table using livewire from blade view

is the error because table search expects for model to get query from?

**Reply by fariqueparammel** [reply]
> hey am a newbie to laravel and filament i tried the doc but gets the error Table [App\Filament\App\Resources\ProjectsResource\Pages\ManualDataEntry] must have a [query()].
> 
> blade view x-filament-panels::page {{-- use interactwithrecords to interact with db i think --}} {{-- {{ $getRecord()->name }} --}} {{ $this->table }} {{ $this->form }} </x-filament-panels::page>
> 
> resource\projectResource\page\ManualDataEntry.php
> 
> emptyStateHeading('No posts yet') ->columns([ TextColumn::make('currency') ->money('EUR') ]) ->filters([ // ... ]) ->actions([ // ... ]) ->bulkActions([ // ... ]); } // i want to dynamically add contents including column heading to table using livewire from blade view is the error because table search expects for model to get query from?

oh i found the solution 
as i thought this was the issue 
 ->query(Model::query())
